### PR TITLE
lldap-cli: init at 0-unstable-2024-02-24

### PR DESCRIPTION
**Pull request by nwhirschfeld**
## Description of changes

https://github.com/Zepmann/lldap-cli

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ll/lldap-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+, coreutils
+, gnugrep
+, gnused
+, jq
+, curl
+, makeWrapper
+}:
+stdenv.mkDerivation {
+  pname = "lldap-cli";
+  version = "nightly-2024-02-24";
```
Comment (eclairevoyant) [inline pkgs/by-name/ll/lldap-cli/package.nix:14]:
```suggestion
  version = "0-unstable-2024-02-24";
```

File: pkgs/by-name/ll/lldap-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  patchPhase = ''
+    # fix .lldap-cli-wrapped showing up in usage
+    substituteInPlace lldap-cli \
+      --replace-fail '$(basename $0)' lldap-cli
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp lldap-cli $out/bin/
```
Comment (eclairevoyant) [inline pkgs/by-name/ll/lldap-cli/package.nix:34]:
```suggestion
    install -Dm555 lldap-cli -t $out/bin
```

File: pkgs/by-name/ll/lldap-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    wrapProgram $out/bin/lldap-cli \
+      --prefix PATH : ${lib.makeBinPath [ bash coreutils gnugrep gnused jq curl ]}
+  '';
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/by-name/ll/lldap-cli/package.nix:39]:
```suggestion
  meta = {
```

tracking issue #292468

File: pkgs/by-name/ll/lldap-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      This API is used by an included web-based user interface.
+      Unfortunately, LLDAP lacks a command-line interface,
+      which is a necessity for any serious administrator.
+      LLDAP-CLI translates CLI commands to GraphQL API calls.
+    '';
+    homepage = "https://github.com/Zepmann/lldap-cli";
+    license = licenses.gpl3;
+    maintainers = [ maintainers.nw ];
+    mainProgram = "lldap-cli";
+    platforms = platforms.unix;
```
Comment (eclairevoyant) [inline pkgs/by-name/ll/lldap-cli/package.nix:54]:
```suggestion
    license = lib.licenses.gpl3Only;
    maintainers = [ lib.maintainers.nw ];
    mainProgram = "lldap-cli";
    platforms = lib.platforms.unix;
```

File: pkgs/by-name/ll/lldap-cli/package.nix [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  };
+
+  buildInputs = [ bash coreutils gnugrep gnused jq curl ];
+  nativeBuildInputs = [ makeWrapper ];
+
+  patchPhase = ''
+    # fix .lldap-cli-wrapped showing up in usage
+    substituteInPlace lldap-cli \
+      --replace-fail '$(basename $0)' lldap-cli
+  '';
```
Comment (eclairevoyant) [inline pkgs/by-name/ll/lldap-cli/package.nix:30]: please keep pre/post-phase hooks. same idea with `installPhase`.
```suggestion
  patchPhase = ''
    runHook prePatch

    # fix .lldap-cli-wrapped showing up in usage
    substituteInPlace lldap-cli \
      --replace-fail '$(basename $0)' lldap-cli

    runHook postPatch
  '';
```

File: pkgs/by-name/ll/lldap-cli/package.nix [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
```
Comment (eclairevoyant) [inline pkgs/by-name/ll/lldap-cli/package.nix:31]:
```suggestion

dontConfigure = true;
dontBuild = true;

```

File: pkgs/by-name/ll/lldap-cli/package.nix [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  version = "nightly-2024-02-24";
+
+  src = fetchFromGitHub {
+    owner = "Zepmann";
+    repo = "lldap-cli";
+    rev = "d1fe50006c4a3a1796d4fb2d73d8c8dcfc875fd5";
+    hash = "sha256-ZKRTYdgtOfV7TgpaVKLhYrCttYvB/bUexMshmmF8NyY=";
+  };
+
+  buildInputs = [ bash coreutils gnugrep gnused jq curl ];
```
Comment (eclairevoyant) [inline pkgs/by-name/ll/lldap-cli/package.nix:23]:
```suggestion
```
these don't do anything as you're manually wrapping, and there's no hook here AFAICT that would use this attr

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! Some feedback below:

File: pkgs/by-name/ll/lldap-cli/package.nix (hunk 7 above)
Comment (nwhirschfeld) [reply pkgs/by-name/ll/lldap-cli/package.nix]: Thanks a lot for your feedback. I'm new to this ecosystem and I'm very happy to improve my contributions over time.
This issue appeared, as I blindly followed the example for shell scripts in the wiki over [here](https://nixos.wiki/wiki/Shell_Scripts#Packaging).

**Review by eclairevoyant — APPROVED** [review]
`--help` at least works, though I'm not familiar with lldap to test this further

File: pkgs/by-name/ll/lldap-cli/package.nix (hunk 7 above)
Comment (eclairevoyant) [reply pkgs/by-name/ll/lldap-cli/package.nix]: For future reference, the wiki is often out-of-date, though this is a good first PR.

**Review by vifino — APPROVED** [review]

**Pull request merged**